queries.sql

```diff
@@ -23,4 +23,88 @@ SELECT * FROM animals WHERE animal_name Not IN ('Gabumon');
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that 
 -- equals precisely 10.4kg or 17.3kg)
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3
+
+
+-- Transactions
+
+-- Inside a transaction update the animals table by setting the species column to unspecified
+BEGIN TRANSACTION;
+UPDATE animals
+SET species ='unspecified';
+
+-- To Rollback
+ROLLBACK;
+
+-- Update the animals table by setting the species column to digimon for all 
+-- animals that have a name ending in mon.
+BEGIN TRANSACTION;
+UPDATE animals
+SET species = 'digimon'
+WHERE animal_name LIKE '%mon';
+
+-- Update the animals table by setting the species column to pokemon for all 
+-- animals that don't have species already set.
+UPDATE animals
+SET species = 'pokemon'
+WHERE animal_name NOT LIKE '%mon';
+
+-- Commit the transaction
+COMMIT;
+
+-- Inside a transaction delete all records in the animals table
+BEGIN TRANSACTION;
+DELETE from animals;
+
+-- To Rollback
+ROLLBACK;
+
+-- Inside a transaction: 
+-- Delete all animals born after Jan 1st, 2022.
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+-- Create a savepoint for the transaction.
+SAVEPOINT animals_savepoint;
+
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+-- Rollback to the savepoint
+ROLLBACK TO animals_savepoint;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+-- Commit the transaction
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(animal_name) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(animal_name) FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attempts)
+FROM animals
+GROUP BY neutered
+ORDER BY SUM(escape_attempts) DESC;
+-- Neutered animals escape the most
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT Min(weight_kg), MAX(weight_kg) FROM animals;
+
+-- Min-Weight is 5.kg and Max-weight is 45kg
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT animal_name, AVG(escape_attempts)
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY animal_name;
```

Comment:
- [ ] This query seeks to find the average number of escape attempts **_per animal_** type born between 1990 - 2000. You have written a query for the animals themselves which is incorrect. The reference **_per animal_** type means per `species`. Kindly update your query to produce the correct result. Please attach the correct screenshot for this query when you are through. 👍 😎